torchdata/dataloader2/README.md

```diff
@@ -0,0 +1,68 @@
+# DataLoader2 (Prototype)
+
+`DataLoader2` is introduced to execute `DataPipe` graph, with support for
+
+- dynamic sharding for multi-process and distributed data loading
+- multiple backend `ReadingServices` (WIP for `DistributedReadingService`)
+- `DataPipe` graph in-place modification like shuffle control, memory pinning, etc.
+- snapshot the state of data-preprocessing pipeline (WIP)
+
+## Why DataLoader2
+
+`DataPipe` is introduced to not only decompose the data pre-processing operations, but also decouple the overloaded
+data-manipulation features from `DataLoader`. So, a light-weight `DataLoader2` is required. Besides, a certain features
+can only be achieved with `DataLoader2` like snapshotting and bridging data-processing pipeline between research and
+production.
+
+These options are configured by the constructor arguments of a `DataLoader2`, which has signature:
+
+```py
+DataLoader2(datapipe, datapipe_adapter_fn=None, reading_service=None)
+```
+
+The sections below describe in details the effects and usages of these options.
+
+## DataPipe
+
+`DataLoader2` supports both `MapDataPipe` and `IterDataPipe`. To understand the basic structure of `DataPipe`, please
+see [What are DataPipes?](https://github.com/pytorch/data#what-are-datapipes)
+
+## Data-Processing Options (datapipe_adapter_fn)
+
+`Adapters` are used to configure, modify and extend the graph of `DataPipes` in `DataLoader2`. For example,
+`Shuffle(False)` can be provided to `DataLoader2`, which would disable any of `shuffle` operations in the graph of
+`DataPipes`.
+
+And, we will provide more `Adapters` to cover data-processing options such as `PinMemory`. If you have feature requests
+about the `Adapters` you'd like to be provided, please open a GitHub issue.
+
+For specific needs, `DataLoader2` accepts any custom `Adapter` as long as it inherits from the `Adapter` class.
+
```

Comment:
Do we have more potentially `Adapter` aside from `Shuffle` and `PinMemory`? I think it will help users to understand the concept better if we give more potential use cases.